tests/integration/test_concurrent_agent_creation.py

```
@@ -0,0 +1,307 @@
+
+    @pytest.mark.slow
+    def test_concurrent_creation_no_corruption(self):
+        """Multiple simultaneous creations don't corrupt the launcher."""
+        with tempfile.TemporaryDirectory() as tmpdir:
+            tmp_path = Path(tmpdir)
+            setup_temp_project(tmp_path)
+
+            results = []
+            num_agents = 5
+
+            def create_agent(i: int):
```

Ruff (0.14.14)

[warning] Missing return type annotation for private function `create_agent`

(ANN202)